source/agora/consensus/EnrollmentManager.d

```diff
@@ -471,6 +471,33 @@ public class EnrollmentManager
         return this.validator_set.hasPreimage(enroll_key, distance);
     }
 
+    /***************************************************************************
+
+        Generate the random seed reduced from the preimages for the provided
+        block height.
+
+        Params:
+            keys = the keys to look up
+            height = the desired block height to look up the images in
+
+        Returns:
+            the random seed
+
+    ***************************************************************************/
+
+    public Hash getRandomSeed (in Hash[] keys, in Height height) @safe nothrow
+    {
+        Hash rand_seed;
+        foreach (const ref key; keys)
+        {
+            const preimage = this.validator_set.getPreimageAt(key, height);
+            assert(preimage != PreImageInfo.init);  // this should not happen
+            rand_seed = hashMulti(rand_seed, preimage);
```

Comment:
Technically this does `hashMulti(Hash.init, preimage)` on the first loop iteration. However I don't think it really matters.